sdk/python/lib/pulumi/resource.py

```diff
@@ -654,8 +661,9 @@ def __init__(self,
 
             opts.aliases = opts.aliases.copy()
             for parent_alias in opts.parent._aliases:
-                opts.aliases.append(inherited_child_alias(
-                    name, opts.parent._name, parent_alias, t))
+                child_alias = inherited_child_alias(
+                    name, opts.parent._name, parent_alias, t)
+                opts.aliases.append(cast(Output[Union[str, Alias]], child_alias))
```

Comment:
We should change this in `ouput.py`:

```
T = TypeVar('T')
```

To this:

```
T = TypeVar('T', covariant=True)
```

And add a comment that `Output[T]` is covariant in T.  

After making that change, I'm 99% sure the refactoring/cast above will not be needed (and the same may be true for many similar cases throughout this PR).